proofs/tree.go

```diff
@@ -312,6 +318,13 @@ func (doctree *DocumentTree) AddLeaf(leaf LeafNode) error {
 	if doctree.filled {
 		return errors.New("tree already filled")
 	}
+	for _, item := range doctree.leaves {
+		if (bytes.Compare(item.Property.CompactName(), leaf.Property.CompactName()) == 0) ||
+			(item.Property.ReadableName() == leaf.Property.ReadableName()) {
+
```

Comment:
Readable name check should be sufficient I believe since it should be unique.

Reply:
@vedhavyas  I think we should check both readable name and compact,  that is reason  why this fix needs two unit test guards : one for readable name and one for compact.  Because `AddLeaf `and `AddLeaves` are related to dynamic creation of documents. Clients can trigger any illegal usage. If we just check readable name , then test case `TestTree_AddTwoLeavesWithSameCompactName` will fail

Reply:
any feedback on this comment? @vedhavyas @lucasvo

Reply:
@zyfrank I think being conservative is better. Your solution is good.